include/seqan3/core/type_traits/function.hpp

```diff
@@ -12,7 +12,50 @@
 
 #pragma once
 
-#include <seqan3/core/platform.hpp>
+#include <functional>
+
+#include <seqan3/core/type_list/traits.hpp>
+
+namespace seqan3
```

Comment:
Why is this not detail like most of the core utilities?

Reply:
Because it can be very useful in many other contexts as well and it works on an entity of the STL and not on seqan3 types.